### PR TITLE
IGNITE-21973: Extend test coverage for SQL F201(CAST function)

### PR DESCRIPTION
**Pull request by lowka**
- Adds sqllogic tests for CAST function except for multi field INTERVAL types (such as YEAR TO MONTH).
- Casts to INTERVAL types from `a type under test` are not included . There are just too many of them and there is already type system level tests for these see `CastResolutionTest`.
- Adds test cases for  INTERVAL DAYS and INTERVAL YEARS. 

https://issues.apache.org/jira/browse/IGNITE-21973

---

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/integrationTest/sql/cast/test_cast_interval_day.test [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# name: test/sql/cast/test_cast_interval_day.test
```
Comment (korlov42) [inline modules/sql-engine/src/integrationTest/sql/cast/test_cast_interval_day.test:1]: let's add a few test cases for complex intervals (like DAYS TO MINUTES). The same for YEAR-MONTH intervals

File: modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# name: test/sql/cast/test_cast_bigint.test
+# description: CAST BIGINT
+# feature: SQL F201(CAST function)
+# group: [cast]
+
+statement ok
+SELECT CAST('100' AS BIGINT);
```
[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:7]: Fixed.

[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:7]: See test_exponent_in_cast

[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:7]: Fixed.

[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:7]: Done.

[the remark this answers is not in the document]
Comment (korlov42) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:7]: first, you haven't address all cases I asked you about (cast from different numerics to target type). second, why do we still have `test_exponent_in_cast`? We have set if tests for particular target type. Let's just inline all these files with test cases from `test_exponent_in_cast `

File: modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test [hunk 3]
```diff
@@ -6,6 +6,10 @@
 statement ok
 SELECT CAST('100' AS BIGINT);
 
+skipif ignite3
```
Comment (korlov42) [inline modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:9]: why do we ignore this only for BIGINT type?

File: modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test (hunk 2 above)
[the remark this answers is not in the document]
Comment (korlov42) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:7]: looks like you've missed cast from interval to BIGINT

[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:7]: Removed test_exponent_in_cast.
Added exponent notations tests for each numeric type.
Added test covering conversion between numeric types.

[the remark this answers is not in the document]
Comment (lowka) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:7]: Thanks. Fixed.

File: modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test (hunk 3 above)
Comment (lowka) [reply modules/sql-engine/src/integrationTest/sql/cast/test_cast_bigint.test:10]: Filed an issue.

**Pull request merged**